extra/mariabackup/xtrabackup.cc

```diff
@@ -4204,11 +4198,11 @@ xb_register_filter_entry(
 
 		if (databases_hash && databases_hash->array) {
 			const ulint fold = my_crc32c(0, dbname, p - name);
-			HASH_SEARCH(name_hash, databases_hash,
-					fold,
-					xb_filter_entry_t*,
-					db_entry, (void) 0,
-					!strcmp(db_entry->name, dbname));
+			db_entry = databases_hash->cell_get(fold)
+				->find<xb_filter_entry_t>(
+					&xb_filter_entry_t::name_hash,
+					[&dbname](xb_filter_entry_t* f)
+					{ return !strcmp(f->name, dbname); });
 		}
 		if (!db_entry) {
 			db_entry = xb_add_filter(dbname, databases_hash);
```

Comment:
This logic is still too complex, and apparently incorrect after this change, because many `--suite=mariabackup` tests would fail. `databases_hash` can never be `nullptr` here; `xb_add_filter()` would have crashed if it were.

I will rewrite this so that we do a `search` followed by a conditional append (in case no entry was found).